Tests/MongoSwiftSyncTests/SpecTestRunner/SpecTest.swift

```diff
@@ -184,13 +184,13 @@ extension SpecTestFile {
 
     func terminateOpenTransactions(
         using client: MongoSwiftSync.MongoClient,
-        mongosClients: [MongoSwiftSync.MongoClient]?
+        mongosClients: [ServerAddress: MongoSwiftSync.MongoClient]
```

Comment:
starting here and in the files below, I make some changes I laid out in the ticket description for SWIFT-1324: https://jira.mongodb.org/browse/SWIFT-1324

in summary, we now use the per-mongos clients (which we already had around for running `killAllSessions` and the "distinct hack") for targeted failpoints, which means the setup client can now just connect to a single mongos. 

we need the setup client to connect to a single mongos when in LB mode because if we use the LB connected to multiple mongoses then we cannot control which mongos a failpoint is set on. I considered a less invasive approach, where when connected to a load balancer we would just always use a single mongos for the setup client and use multiple mongoses for other topologies, but this seemed like a better solution overall as it helps with test flakiness issues too.